smart_open/tests/test_smart_open.py

```diff
@@ -287,6 +289,35 @@ def test_gs_uri_contains_slash(self):
         self.assertEqual(parsed_uri.bucket_id, "mybucket")
         self.assertEqual(parsed_uri.blob_id, "mydir/myblob")
 
+    def test_pathlib_monkeypath(self):
+        assert pathlib.Path.open != smart_open.open
+
+        with patch_pathlib():
+            assert pathlib.Path.open == smart_open.open
+
+        assert pathlib.Path.open != smart_open.open
+
+        obj = patch_pathlib()
+        assert pathlib.Path.open == smart_open.open
+
+        _patch_pathlib(obj.old_impl)
+        assert pathlib.Path.open != smart_open.open
+
+    def test_pathlib_monkeypath_read_gz(self):
+        path = pathlib.Path(CURR_DIR) / 'test_data' / 'crime-and-punishment.txt.gz'
+
+        # Check that standart implementation can't work with gzip
+        with path.open("r") as infile:
+            with self.assertRaises(Exception) as context:
+                lines = infile.readlines()
+
+        # Check that out implementation works with gzip
+        obj = patch_pathlib()
+        with path.open("r") as infile:
+            lines = infile.readlines()
+
+        _patch_pathlib(obj.old_impl)
+
```

Comment:
Can you add a test for the context manager? That should be the preferred way of restoring the original, instead of calling `_patch_pathlib(obj.old_impl)`.

Reply:
I added context manager for testing purposes, I see no cases for "real usage" of context manager in this case